2. Xizmatlar/2.4 Aggregatsiyalar/2.4 Aggregatsiyalar.md

```diff
@@ -0,0 +1,17 @@
+<h1>2.4 Aggregatsiya(To'plash) Xizmatlari </h1>
+<h3>2.4.0 Kirish</h3>
+Aggregatsiya xizmatining asosiy vazifasi biznes mantiq qatlami o'zaki va tashqi qatlamlari orasida bitta bog'liqlikni ko'rsatish. Bu har qanday o'zgarishlarning bir nechta xizmatlari, bir xil shartnoma asosida to'planishi va bitta taqdim qiluvchi tarkibiy qismga, bitta mantiqiy qatlam orqali oshkor qilinishini(ko'rsatilishini) ta'minlaydi.
+
+Aggregatsiya xizmatlari o'zlarida hech qanday biznes mantig'iga ega emas. Ular shunchaki to'qigich kabidir, bir qancha xizmatlar birga tugulgan go'yo. 
+
+```csharp
+public async ValueTask ProcessStudentAsync(Student student)
+{
+    await this studentRegistrationCoordinationService.RegisterStudentAsync(student);
+    await this.studentRecordsCoordinationService.AddStudentRecordAsync(student);
+    ...
+    ...
+    await this.anyOtherStudentRelatedCoordinationService.DoSomethingWithStudentAsync(student);    
+}
+```
+Tepadagi koddagi kabi Aggregatsiya(To'plash) Xizmati bir necha marotaba har qanday tartibsiz va  cheklovlarsiz chaqirilishi mumkin. Va bu yerda ba'zi holatlarda, taqdimot qatlamlari umumiy tuzulishi va arxitekturasiga qarab, qiymat qaytarishingiz yoki qaytarmasligingiz mumkin,buni qisqacha shu sahifada muhokama qilamiz. Ammo, eng muhimi, Aggregatsiya xizmatlarini  Orchestratsiya va uning xizmat turlari bilan adashtirmaslik kerak.
```

Comment:
Yuqorida ko'rsatilganidek, Aggregatsiya(To'plash) xizmati chegarasiz har qanday buyruqlar sonini qabul qilishi mumkin. Va bu yerda ba'zi holatlarda, umumiy bo'lgan jarayon va arxitekturaga bog'liq ravishda, taqdim etish qatlamlariga qiymat qaytarishingiz yoki qaytarmasligingiz mumkin, biz buni qisqacha ushbu bobda muhokama qilamiz. Ammo, eng muhimi, Aggregatsiya xizmatlarini Orchestratsiya va har qanday boshqa uning xizmatlari bilan  adashtirmaslik kerak.